lib/ui/page/home/page/chat/widget/message_info/controller.dart

```diff
@@ -17,61 +17,122 @@
 
 import 'dart:async';
 
+import 'package:flutter/scheduler.dart';
 import 'package:flutter/widgets.dart';
 import 'package:get/get.dart';
 
 import '/domain/model/chat.dart';
+import '/domain/repository/chat.dart';
 import '/domain/repository/user.dart';
-import '/domain/service/user.dart';
-import '/ui/widget/text_field.dart';
+import '/domain/service/chat.dart';
+import '/util/message_popup.dart';
+import '/util/obs/obs.dart';
 
 /// Controller of the [MessageInfo] popup.
 class MessageInfoController extends GetxController {
-  MessageInfoController(this._userService, {this.reads = const []});
+  MessageInfoController(this.chatId, this.chatService, {this.reads = const []});
+
+  /// ID of the [Chat] this page is about.
+  final ChatId? chatId;
+
+  /// [Chat]s service used to get the [chat] value.
+  final ChatService chatService;
```

Comment:
Why public? I've never seen a service being public in any other controller. Correct me, if I'm wrong.